src/app/components/login/login.component.spec.ts

```diff
@@ -20,4 +33,34 @@ describe('LoginComponent', () => {
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should have a login as administrator anchor', () => {
+    const login: HTMLAnchorElement[] = fixture.nativeElement.querySelectorAll('.btn');
+    const loginAsAdmin = Array.from(login).find((el) => el.textContent === 'Login as Admin');
+    expect(loginAsAdmin).toBeTruthy();
+  })
+
+  it('should have a login as employee anchor', () => {
+    const login: HTMLAnchorElement[] = fixture.nativeElement.querySelectorAll('.btn');
+    const loginAsEmployee = Array.from(login).find((el) => el.textContent === 'Login as Employee');
+    expect(loginAsEmployee).toBeTruthy();
+  });
+
+  it('should emit a call with property isAdmin set to true when login as admin is clicked', () => {
+    const login: HTMLAnchorElement[] = fixture.nativeElement.querySelectorAll('.btn');
+    const loginAsAdmin = Array.from(login).find((el) => el.textContent === 'Login as Admin');
+    expect(loginAsAdmin).toBeTruthy();
+    spyOn(component, 'logIn');
+    loginAsAdmin!.click();
+    expect(component.logIn).toHaveBeenCalledWith(true);
```

Comment:
Looking at this test, I realized that you're having a `boolean` for administrator vs employee. This has two issues:
1. It can become a mess as soon as one more role in the app is introduced
2. It makes it very unreadable looking at the code what this `true` we're passing means.

What we could do instead is have an enum for `UserType` as follows:
```ts
type UserType {
  Admin,
  Employee
}
```

and then here we could check something like:
```ts
// spy on it before
expect(component.loginService.logIn).toHaveBeenCalledWith(UserType.Admin);
```

Reply:
Also, the goal here should be to check if the service's method is called eventually. As we know if we click the button, `component.logIn` should have been called.